server/v2/api/grpcgateway/interceptor.go

```diff
@@ -143,3 +225,26 @@ func getHTTPGetAnnotationMapping() (map[string]string, error) {
 
 	return httpGets, nil
 }
+
+// createRegexMapping converts the annotationMapping (HTTP annotation -> query input type name) to a
+// map of regular expressions for that HTTP annotation pattern, to queryMetadata.
+func createRegexMapping(logger log.Logger, annotationMapping map[string]string) map[*regexp.Regexp]queryMetadata {
+	regexQueryMD := make(map[*regexp.Regexp]queryMetadata)
+	seenPatterns := make(map[string]string)
+	for annotation, queryInputName := range annotationMapping {
+		pattern, wildcardNames := patternToRegex(annotation)
```

Comment:
seems not easy to differentiate uri like `blocks/latest` and `blocks/{height}` with `patternToRegex`

Reply:
fixed it! i just made it so theres a map for simple URI's with no wildcards, and a separate map for complex/wildcard queries. 

the matching logic will always check if we can get away with the simple query first, then it will try the regex matchers.